### DISCUSSION TITLE
[Help] Error in generation pipeline

### DISCUSSION THREAD

**Discussion by kwcckw** [aub-mind/arabert#110] [Q&A] [answered]
Hi, 

Good day.

I'm facing error below when trying to generate text with ara-gpt2:

![image](https://user-images.githubusercontent.com/52521979/120779738-f3530e80-c559-11eb-86ed-8680f853d6c8.png)

And here's the code:

    import torch
    
    from transformers import GPT2TokenizerFast, pipeline
    #for base and medium
    from transformers import GPT2LMHeadModel
    #for large and mega
    from aragpt2.grover.modeling_gpt2 import GPT2LMHeadModel
    
    from preprocess import ArabertPreprocessor
    
    device = 0 if torch.cuda.is_available() else -1
    
    MODEL_NAME='aragpt2-base'
    arabert_prep = ArabertPreprocessor(model_name=MODEL_NAME)
    
    text1  = 'ذهب نوح مظفر ضرغام بصحبة رؤوف بن لؤي ظافر عطعوط وهلال  ';
    text2 = ' خازن عفيف للحج';
    
    text_clean = arabert_prep.preprocess(text1)
    
    model = GPT2LMHeadModel.from_pretrained(MODEL_NAME)
    tokenizer = GPT2TokenizerFast.from_pretrained(MODEL_NAME)
    generation_pipeline = pipeline("text-generation",model=model,tokenizer=tokenizer, device=device)
    
    input_size = len(tokenizer.tokenize(text_clean))
    
    gen_text = generation_pipeline(text_clean,
                pad_token_id=0, # 0 for AraGPT2
                num_beams=5,
                max_length=text_clean+1,
                top_p=0.95,
                repetition_penalty = 3.0,
                no_repeat_ngram_size = 3)[0]['generated_text']
                
Could you advise on any troubleshooting steps? At this point i'm quite clueless on the error.

Thank you.

**Comment by WissamAntoun** [comment, marked as answer]
check this notebook with has the code running for all AraGPT2 models https://colab.research.google.com/drive/1T5xKRAtKeuUmmokyThvkK8pulHw02CoK?usp=sharing#scrollTo=hDeZLZxypXGQ

I think the issue in the code above is that you are using the `GPT2LMHeadModel` class from our repository with the `base` model, instead of using the one provided by Huggingface

**Reply by kwcckw** [reply]
Hi Wissam Antoun,

Yes, i did following that colab file and i downloaded the files from Huggingface:

from Huggingface:
![image](https://user-images.githubusercontent.com/52521979/120792274-d45b7900-c567-11eb-816a-efb5032864ad.png)

The one in local disk:
![image](https://user-images.githubusercontent.com/52521979/120792301-e3422b80-c567-11eb-92c7-22eb75880357.png)

I'm downloaded the files from this link:
https://huggingface.co/aubmindlab/aragpt2-base/tree/main

If this is not the correct path, could you point me to the correct path?

Thank you.

**Reply by WissamAntoun** [reply]
Yes this is the correct repository.
To use the model just provide the path of the folder you just downloaded to model parameter in the pipeline class.

**Reply by kwcckw** [reply]
I tried to redownload the whole repo and run with the sample code, but this time i'm getting another error:

![image](https://user-images.githubusercontent.com/52521979/120876102-74091d80-c5e1-11eb-809c-32e7ef986805.png)

Probably is due to version mismatch issue. Is it possible for you provide the complete list of requirements?

Right now i can see there are only 3 libraries in the requirement file:


        #tensorflow &gt;= 1.11.0, &lt;2.0   # CPU Version of TensorFlow.
      
      
        # tensorflow-gpu  &gt;= 1.11.0, &lt;2.0 # GPU version of TensorFlow.
      
      
        PyArabic
      
      
        farasapy

Thank you.

**Reply by kwcckw** [reply]
Looks like i resolved the issue by reinstalling pip install transformers==4.2.1, the newer version of transformer will facing error as above.

Thanks again for your help and appreciate it.
